linode/manager
# feat: [M3-7328] - Add 'Reboot Needed' status for Linodes assigned to VPCs

## Description 📝
- Add the 'Reboot Needed' status to the VPC details page and Linode Entity header page

## Changes  🔄
List any change relevant to the reviewer.
- Add reboot needed status to VPC details and linode details pages + add logic to make sure it updates
- Added aria-labels to status icons 
- Add inline 'Reboot' action menu item for VPC details page if Linode is active and needs reboot; otherwise show 'Power On' inline action menu if the Linode is inactive
- update tests
- Make VPC details banner universal, not unique for each subnet
- Some random eslint fixes on files that I was looking through while working on this

## Preview 📷

| Before  | After   |
| ------- | ------- |
| ![image](https://github.com/linode/manager/assets/139280159/a57b11cc-4c6b-40ac-b4b0-e36d90c1440d) | ![image](https://github.com/linode/manager/assets/139280159/2139a2fd-a4ed-4255-bf70-7e9b16f8732d) |

| Before  | After   |
| ------- | ------- |
| ![image](https://github.com/linode/manager/assets/139280159/1bef25c2-32fa-4bc3-8102-d4868b8c83f8) | ![image](https://github.com/linode/manager/assets/139280159/58492c16-6989-4e64-8793-bd81d7794a4c) |


## How to test 🧪

### Prerequisites
(How to setup test environment)
- With VPC account tags, and with at least one vpc with a subnet and a linode in that same region,

### Verification steps 
(How to verify changes)
On the VPC's detail pages
- Assign an already running linode a subnet. Confirm the reboot status and help icon show up, and that the reboot linode button appears
- Reboot the linode from the inline action menu. Confirm that when the linode is no longer rebooting, the status no longer says 'Reboot needed'
- navigate to the Linode's detail page and confirm status matches the subnet detail page
- If the Linode assigned to a subnet is offline, then there should be a 'Power On' inline button instead of a 'Reboot' button

NOTE: As of now, the Reboot Needed status only makes sense for linodes with one config. Statuses for Linodes with multiple configs assigned to subnets may be very confusing -- see internal ticket for details

## As an Author I have considered 🤔

*Check all that apply*
